### ISSUE TITLE
[BUG] cveUMatCreate does not pass UMatUsageFlags flags

### ISSUE THREAD

**Issue by odalet** [emgucv/emgucv#762]
**Describe the bug**
I'm confused by the way an `umat` is created. On the C# side (UMat.cs) I read:

```csharp
public UMat() : this(UMatInvoke.cveUMatCreate(), true) { }
...
[DllImport(CvInvoke.ExternLibrary, CallingConvention = CvInvoke.CvCallingConvention)]
internal static extern IntPtr cveUMatCreate();
```

Then, On the C side (umat_c.cpp/.h):

```cpp
public UMat() : this(UMatInvoke.cveUMatCreate(), true) { }
...
CVAPI(cv::UMat*) cveUMatCreate(cv::UMatUsageFlags flags); // .h

cv::UMat* cveUMatCreate(cv::UMatUsageFlags usage) // .cpp
{
   return new cv::UMat(usage);
}
```

How come the flags parameter is not declared on the C# side? I'm not even sure how this can even work: is whatever is in memory at the time of the call just passed down to the C side? Does that mean `UMatUsageFlags` is randomly set to anything?

I'm raising this issue not only for the sake of understanding how the code works, but also because 
I experience weird (and not easily reproductible) behavior when I use `Umat` with OpenCL **disabled**. Sometimes, it appears that, although I explicitly call `CvInvoke.UseOpenCL = false` beforehand, `UseOpenCL` sometimes is `true` again. I've managed to trace this behavior around the creation of `Umat` objects and this flag that is *not* passed when creating the `UMat` may have a role in enabling or not OpenCL...

** OS / Platform **
all platforms.

** .Net version **
any

**CPU Architecture**
x86_64

** Emgu CV package used**
Emgu.CV.Platform.NetStandard, Version=4.5.5.4823

**Expected behavior**
I expect that when `CvInvoke.UseOpenCL is `false` it remains so throughout the execution of my program
Additionally, I'd appreciate having a means to pass `cv::UMatUsageFlags`  when creating a new `UMat`.

**Comment by emgucv** [comment]
Fixed in the above commit. Thanks for pointing out.

**Comment by odalet** [comment]
Thanks for the fix; any idea when v4.5.6 will be released though?

**Comment by emgucv** [comment]
It is in the 4.6.0 release. Closing ticket now.
